managed/services/checks/checks.go

```diff
@@ -1266,8 +1279,18 @@ WHERE datallowconn = true AND datistemplate = false AND has_database_privilege(c
 		return nil, errors.Wrapf(err, "failed to select available databases")
 	}
 
+	dec, err := b64.DecodeString(res)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode database discovery results")
+	}
+
+	data, err := agentpb.UnmarshalActionQueryResult(dec)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshall database discovery results")
+	}
```

Comment:
This block is similar to the `unmarshalQueryResult` function in `main.go`. I wonder if we can export bring the function to this package and export it for `main.go` to use.

Reply:
I will address these comments in another PR. For now we need just to fix RC ASAP.